test/validation/api/ipsec/ipsec.c

```diff
@@ -1115,16 +1115,19 @@ int ipsec_init(odp_instance_t *inst, odp_ipsec_op_mode_t mode)
 		fprintf(stderr, "Packet pool creation failed.\n");
 		return -1;
 	}
-	if (mode == ODP_IPSEC_OP_MODE_ASYNC ||
-	    mode == ODP_IPSEC_OP_MODE_INLINE) {
+
+	if (suite_context.inbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
+	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
+	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_INLINE) {
```

Comment:
If inbound_op_mode is inline and outbound_op_mode is sync, this function fails to create the default queue.

Reply:
Now when I look at this again, from the spec what I understand is that, default_queue will be used only where inbound lookup failure happens. And that is dependent only on the inbound_op_mode. So I think the following if conditional would be enough for the default_queue creation:

```
if (suite_context.inbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
   suite_context.inbound_op_mode == ODP_IPSEC_OP_MODE_INLINE) {
   ...
}
```

Please correct me if I am wrong.

Reply:
I think that is correct. The default queue is used when SA is not known (if SA was known, then the SA specific queue would be used), and in outbound processing the SA is always known.